scripts/src/main/resources/scripts/command/intellij

```diff
@@ -65,6 +66,64 @@ function doSetup() {
   fi
 }
 
+function doAddPlugins() {
+  local file
+  for file in "${SETTINGS_PATH}"/intellij/plugins/*.properties
+  do
+    if [ -f "${file}" ]
+    then
+      plugin_id=""
+      plugin_active="true"
+      doLoadProperties "${file}"
+      if [ -z "${plugin_id}" ]
+      then
+        doWarning "Invalid intellij plugin config: ${file}"
+      elif [ "${plugin_active}" = "true" ]
+      then
+        doAddPlugin "${plugin_id}"
+      fi
+    fi
+  done
+}
+
+function doAddPlugin() {
+  local intellij_build
+  local download_dir="${DEVON_DOWNLOAD_DIR}"
+  local filename
+  local install_path="${IDEA_HOME}/plugins"
+
+  mkdir -p "${install_path}"
+
+  if [ -e "${IDEA_HOME}/build.txt" ]
+  then
+    intellij_build="$(cat "${IDEA_HOME}/build.txt")"
+  elif [ -e "${IDEA_HOME}/IntelliJ.app/Contents/Resources/build.txt" ]
+  then 
+    intellij_build="$(cat "${IDEA_HOME}/IntelliJ.app/Contents/Resources/build.txt")"
+  fi
+
+  if [ -n "${1}" ] && [ -n "${intellij_build}" ]
+  then
+    local download_url="https://plugins.jetbrains.com/pluginManager?action=download&id=${1}&build=${intellij_build}"
+    filename=$(curl -sI "${download_url}" | grep -o -E 'location:.*$' | sed -e 's/location://')
+    filename="${filename##*/}"
+    filename="${filename%%\?*}"
+    (cd "${download_dir}" && curl -OJ --location-trusted "${download_url}")
```

Comment:
cant we recycle some logic from the doDownload function? There is quite some shared logic in there including to choose the proper directory to download to and also the downloading tool. We already considered supporting wget as fallback if curl is not installed, etc. Such features are only possible if we keep the logic centralized in functions. 
Cant we just create a filename from `$1` and provide it as arg? If needed we can also extend the doDownload function but IMHO you can already provide filename, etc. as parameters to make it fit for your use-case.

Reply:
I had first tried to do it with `doDownload`. The problem here is that I not have the filename. With `-OJ --location-trusted` I wanted to download the file with the correct name, that works, but then I noticed that I have to check the extension whether it is zip or jar. Because a plugin can be a jar or zip and only the zip should be unzipped. I missed that now I get the filename before, and now I can use `doDownload`.